CRM/Financial/BAO/Payment.php

```diff
@@ -99,6 +99,9 @@ public static function create($params) {
         CRM_Contribute_BAO_Contribution::assignProportionalLineItems($params, $trxn->id, $contribution['total_amount']);
       }
     }
+    elseif ($params['total_amount'] < 0) {
```

Comment:
@monishdeb I've updated this PR with this call to record refunds from here - hopefully that means payment_cancel can call Payment.create now & we are closer to being rid of recordAdditionalPaymen

Reply:
@eileenmcnaughton  Yes, that would be fine.

Also, let me share my thoughts about what I was supposed to do to support refund using payment processor. I am thinking of having a new api payment.refund which will do refund using payment processor and eventually will call Payment.cancel API to record financial records - https://github.com/civicrm/civicrm-core/compare/master...JMAConsulting:financial-38?expand=1#diff-8967f6e57d6209aa277ded6512413608R510

Reply:
And then for refund w/o PP (i.e. Manual PP) the only call to payment.cancel will do.

Reply:
@monishdeb so I'm not sold on having one api do both the payment processor refund AND process the refund payment. I think we could sink into various edge cases so I lean towards having

PaymentProcessor.pay (already in Omnipay) and PaymentProcessor.refund (which would call doRefund on the processor class) and then the calling code can call either Payment.create or Payment.cancel - I'm not sure which makes more sense but they will eventually be more or less the same code anyway

Reply:
(I also think we can add PaymentProcessor.refund first & then as we start to use it we can consider if we have needs not met by the Payment apis & the PaymentProcessor.refund api rather than decide that up front - as the work for PaymentProcessor.refund will be required regardless